### PR TITLE
Add and update rules for RETURN and BREAK,CONTINUE statements

### PR DESCRIPTION
**Pull request by bhirsz**
* add ``return-in-test-case`` rule
* update (new) deprecated-statement rule to also catch usage of ``Exit For Loop``, ``Exit For Loop If``, ``Continue For Loop``, ``Continue For Loop If``, ``Return From Keyword``. ``Return From Keyword If`` and``[Return]`` (instead of ``RETURN``)

### CODE REVIEW

**Comment by bhirsz** [comment]
I will also update ``keyword-after-return`` rule inside this PR before the release (support also ``RETURN``). Docs for this rule will need some adjustments to be comprehensive for both <5 and >=5 RF users.

File: robocop/checkers/naming.py [hunk 1]
```diff
@@ -676,55 +677,51 @@ def find_similar_variables(self, tokens, node):
...
-    def visit_TestSetup(self, node):  # noqa
-        self.check_if_keyword_is_deprecated(node.name, node)
-
-    def visit_Setup(self, node):  # noqa
-        self.check_if_keyword_is_deprecated(node.name, node)
-
-    def visit_SuiteTeardown(self, node):  # noqa
-        self.check_if_keyword_is_deprecated(node.name, node)
+    reports = ("deprecated-statement",)
+    depreccated_st = {5: {"[Return]"}}
```
Comment (mnojek) [inline robocop/checkers/naming.py:681]: I would put a longer and more descriptive name for this variable. (also there was a typo)
```suggestion
    deprecated_statements = {5: {"[Return]"}}
```

File: robocop/checkers/naming.py [hunk 2]
```diff
@@ -676,55 +677,51 @@ def find_similar_variables(self, tokens, node):
...
+    deprecated_kws = {
```
Comment (mnojek) [inline robocop/checkers/naming.py:682]:
```suggestion
    deprecated_keywords = {
```

File: robocop/checkers/naming.py [hunk 3]
```diff
@@ -676,55 +677,51 @@ def find_similar_variables(self, tokens, node):
 class DeprecatedStatementChecker(VisitorChecker):
     """Checker for deprecated statements."""
 
-    reports = (
-        "deprecated-statement",
-    )
-    # deprecated:alternative
-    deprecated_statements_rf3 = {}
-    deprecated_statements_rf4 = {"runkeywordunless": "IF"}
```
Comment (mnojek) [inline robocop/checkers/naming.py:684]: So this rule should no longer be supported for RF 4?

File: robocop/checkers/naming.py [hunk 4]
```diff
@@ -676,55 +677,51 @@ def find_similar_variables(self, tokens, node):
...
+        }
+    }
 
-    def visit_TestTeardown(self, node):  # noqa
-        self.check_if_keyword_is_deprecated(node.name, node)
+    def visit_KeywordCall(self, node):  # noqa
+        self.check_if_keyword_is_deprecated(node.keyword, node)
 
-    def visit_Teardown(self, node):  # noqa
+    def visit_SuiteSetup(self, node):
```
Comment (mnojek) [inline robocop/checkers/naming.py:698]: I think you missed `# noqa` here

**Review by mnojek — CHANGES_REQUESTED** [review]

File: robocop/checkers/naming.py (hunk 3 above)
Comment (bhirsz) [reply robocop/checkers/naming.py:684]: Good question actually - for Run Keyword If there is ``if-can-be-used`` rule (that works only for ==4.*). There is nothing for ``Run Keyword Unless``. So I can at least make this rule work for 4.0

**Comment by bhirsz** [comment]
@mnojek I've also updated keyword-after-return extensively. It now supports ``RETURN`` statement in RF 5.0. I've also decided to remove ``keyword-after-return-from`` rule and merge it's behaviour into ``keyword-after-return``. It was making sense to have two separate rules before (kind of..) but now it's better to use one rule. WDYT?

File: robocop/checkers/misc.py [hunk 5]
```diff
@@ -270,34 +269,49 @@ class ReturnChecker(VisitorChecker):
...
         return_setting_node = None
         keyword_after_return = False
         return_from = False
+        error = ""
         for child in node.body:
             if isinstance(child, Return):
                 return_setting_node = child
+                error = (
+                    "[Return] is not defined at the end of keyword. "
+                    "Note that [Return] does not return from keyword but only set returned variables"
```
Comment (mnojek) [inline robocop/checkers/misc.py:285]: I guess that would make more sense:
```suggestion
                    "Note that [Return] does not quit from keyword but only set variables to be returned"
```

File: robocop/checkers/misc.py [hunk 6]
```diff
@@ -270,34 +269,49 @@ class ReturnChecker(VisitorChecker):
...
+
+            if isinstance(child, KeywordCall):
                 if return_setting_node is not None:
                     keyword_after_return = True
                 if return_from:
                     token = child.data_tokens[0]
                     self.report(
-                        "keyword-after-return-from",
+                        "keyword-after-return",
+                        error_msg="Keyword call after 'Return From Keyword' keyword",
```
Comment (mnojek) [inline robocop/checkers/misc.py:303]: I think we can safely remove this redundant word here:
```suggestion
                        error_msg="Keyword call after 'Return From Keyword'",
```

**Review by mnojek — APPROVED** [review]

**Review by mnojek — APPROVED** [review]

**Pull request merged**